src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/mob_shattered_hand_legionnaire.cpp

```diff
@@ -0,0 +1,226 @@
+
+                if (Creature* sleeping = GetClosestCreatureWithGuid(m_creature, aSleepingReinfGuid[urand(0, 3)], 150.0f))
+                {
+                    sleeping->GetMotionMaster()->MovePoint(100, FelOrcMoveCoords[legionnaireGuid][0], FelOrcMoveCoords[legionnaireGuid][1], FelOrcMoveCoords[legionnaireGuid][2]);
+                    sleeping->RemoveAurasDueToSpell(16093);
+                }
+                nearbyFriendDied = false;
+            }
+            else
+            { 
+                // Have to reasearch the cooldown for FelOrc spawn
+                if (m_uiMinionSpawnTimmer < uiDiff)
+                {                
+
+                    if (Creature* felorc = m_creature->SummonCreature(MOB_FEL_ORC, FelOrcSpawnCoords[legionnaireGuid][0], FelOrcSpawnCoords[legionnaireGuid][1], FelOrcSpawnCoords[legionnaireGuid][2], FelOrcSpawnCoords[legionnaireGuid][3], TEMPSPAWN_TIMED_OOC_OR_DEAD_DESPAWN, 100000, true, true))
+                    {
+                        felorc->GetMotionMaster()->MovePoint(100, FelOrcMoveCoords[legionnaireGuid][0], FelOrcMoveCoords[legionnaireGuid][1], FelOrcMoveCoords[legionnaireGuid][2]);
+                        felorc->SetRespawnCoord(FelOrcMoveCoords[legionnaireGuid][0], FelOrcMoveCoords[legionnaireGuid][1], FelOrcMoveCoords[legionnaireGuid][2], FelOrcMoveCoords[legionnaireGuid][3]);
+                    }
+                    DoScriptText(aRandomReinf[urand(0, 10)], m_creature);
+                    nearbyFriendDied = false;
+                    m_uiMinionSpawnTimmer = 5000;
+                }
+                else
+                    m_uiMinionSpawnTimmer -= uiDiff;
+            }
+        }
+
+        if (!m_creature->SelectHostileTarget() || !m_creature->GetVictim())
+            return;
+
+        if (m_uiPummelTimmer < uiDiff)
+        {
+            if (Unit* pTarget = m_creature->SelectAttackingTarget(ATTACKING_TARGET_RANDOM, 0, SPELL_PUMMEL, SELECT_FLAG_PLAYER_CASTING))
+            {
+                if (DoCastSpellIfCan(pTarget, SPELL_PUMMEL) == CAST_OK)
+                    m_uiPummelTimmer = urand(10000, 15000);
+            }
+        }
+        else
+            m_uiPummelTimmer -= uiDiff;
+
+        if (m_uiAuraTimer < uiDiff)
+        {
+            m_creature->CastSpell(m_creature, SPELL_AURA_OF_DISCIPLIN, TRIGGERED_NONE);
+            m_uiAuraTimer = 15000;
+        }
+        else
+            m_uiAuraTimer -= uiDiff;
+
+        if (m_creature->GetHealthPercent() <= 30.0f)
+            m_creature->CastSpell(m_creature, SPELL_ENRAGE, TRIGGERED_NONE);
+
+        DoMeleeAttackIfReady();
+    }
+};
+
+UnitAI* GetAI_mob_shattered_hand_legionnaire(Creature* pCreature)
+{
+    return new mob_shattered_hand_legionnaireAI(pCreature);
+}
+
+void AddSC_mob_shattered_hand_legionnaire()
+{
+    Script* pNewScript = new Script;
+    pNewScript = new Script;
+    pNewScript->Name = "mob_shattered_hand_legionnaire";
+    pNewScript->GetAI = &GetAI_mob_shattered_hand_legionnaire;
```

Comment:
do GetNewAIInstance<mob_shattered_hand_legionnaireAI> and remove GetAI_mob_shattered_hand_legionnaire

Reply:
Okay, gonna change it.